compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2623,6 +2623,11 @@ fn add_apple_sdk(cmd: &mut dyn Linker, sess: &Session, flavor: LinkerFlavor) {
         ("x86", "ios") => "iphonesimulator",
         ("x86_64", "ios") if llvm_target.contains("macabi") => "macosx",
         ("x86_64", "ios") => "iphonesimulator",
+        ("x86_64", "watchos") => "watchsimulator",
+        ("arm64_32", "watchos") => "watchos",
+        ("aarch64", "watchos") if llvm_target.contains("sim") => "watchsimulator",
```

Comment:
Preexisting for other targets, but consider `llvm_target.ends_with("simulator")`, and in general use the same logic for all architectures, so

```
(_, "watchos") if llvm_target.ends_with("-simulator") => "watchsimulator",
```